src/app/shared/models/student-grades.ts

```diff
@@ -131,3 +131,35 @@ function isGrade(g: GradeOrNoResult): g is Grade {
 export function toMaxPoints(grade: GradeOrNoResult | null): number {
   return grade?.test.MaxPointsAdjusted || grade?.test.MaxPoints!;
 }
+
+export function meanOf(finalGrades: FinalGrade[]): number {
+  const averageGrades = finalGrades
+    .map((finalGrade) => finalGrade.average)
+    .map((maybeNumber) => maybeNumber?.valueOf())
+
+    .filter(
+      (averageGrade) =>
+        averageGrade !== null &&
+        averageGrade !== undefined &&
+        averageGrade !== 0
+    )
+    .map(Number);
+  return average(averageGrades, 3);
```

Comment:
3: Magic number?

Reply:
jein - ich habs mal so gemacht - ich wusste nicht, dass es die Pipe gibt - so wie es jetzt ist, kann ich hier bestimmen, wie viele Stellen für Punkte und FinalGrades ausgegeben werden. Eigentlich gehört das in die View